src/usecase/client_user_ucase.go

```diff
@@ -65,9 +65,10 @@ func (cu *ClientUser) Add(c *gin.Context) { //nolint:dupl
 	user.Password = utils.HashString(password)
 
 	existingUser, err := userRepo.GetByKey("email", user.Email)
+	var userStatus string
+	userStatus = *existingUser.Status
```

Comment:
don't use var declaration when you can use short declaration
` userStatus := *existingUser.Status`
and I've created util function which called DerefString, you might run into some panics or run time errors when the user won't have any status and your pointer value would be nil, that function handles these cases

Reply:
And you don't to create this variable here since it might be not needed, if user ID is not nil, only in that case you would need user's status